### PR TITLE
Add conditional for enabling/disabling sync dest

### PR DESCRIPTION
**Pull request by yertsti**
# Ticket Link

[1525](https://uhawaii.atlassian.net/browse/GROUPINGS-1525)

# List of squashed commits

- Add conditional for enabling/disabling sync dest

# Test Checklist

- [x] Unit Tests Passed:
- [x] Jasmine Tests Passed:
- [x] General Visual Inspection:

### CODE REVIEW

**Comment by yertsti** [comment]
Is there an easy way to re-open a merged PR?

**Review by mhodgesatuh — APPROVED** [review]

**Comment by fduckart** [comment]
Me, I think it's fine to open a new PR like you did. Maybe we can try an experiment for this that's a little involved. I'm thinking you could make a new custom dialog controller. Slack me so we can brainstorm a bit.

File: src/main/resources/static/javascript/mainApp/grouping.controller.js [hunk 1]
```diff
@@ -1582,7 +1582,12 @@
                 scope: $scope,
                 backdrop: "static",
                 keyboard: false,
-                ariaLabelledBy: "sync-dest-modal"
+                ariaLabelledBy: "sync-dest-modal",
+                controller: "SyncDestModalController",
+                resolve: {
+                    isSynced: isSyncDestOn,
+                    syncDestDescription: () => $scope.selectedSyncDest.description,
+                }
```
Comment (JorWo) [inline src/main/resources/static/javascript/mainApp/grouping.controller.js:1590]: I think this $scope.displaySyncDestModal() function belongs inside the SyncDestModalController.

File: src/test/javascript/grouping.controller.test.js [hunk 2]
```diff
@@ -2711,3 +2683,44 @@ describe("GroupingController", () => {
...
+
+        it("should set the sync dest description", () => {
+            expect(scope.syncDestDescription).toBe("Sync dest description");
+        });
+
+        it("should set the confirmation based on sync state", () => {
+            expect(scope.confirmationMessage).toContain("enable")
+        });
+    })
+});
```
Comment (JorWo) [inline src/test/javascript/grouping.controller.test.js:2726]: Could you fix the EOF error here? Thanks.

File: src/main/resources/static/javascript/mainApp/grouping.controller.js [hunk 3]
```diff
@@ -1582,7 +1582,12 @@
                 scope: $scope,
                 backdrop: "static",
                 keyboard: false,
-                ariaLabelledBy: "sync-dest-modal"
+                ariaLabelledBy: "sync-dest-modal",
+                controller: "SyncDestModalController",
+                resolve: {
+                    isSynced: isSyncDestOn,
+                    syncDestDescription: () => $scope.selectedSyncDest.description,
```
Comment (JorWo) [inline src/main/resources/static/javascript/mainApp/grouping.controller.js:1589]: Does syncDestDescription need to be a function? I'm not sure.

File: src/main/resources/static/javascript/mainApp/grouping.controller.js (hunk 1 above)
Comment (JorWo) [reply src/main/resources/static/javascript/mainApp/grouping.controller.js:1590]: Sorry, you can ignore this comment.

File: src/main/resources/templates/modal/syncDestModal.html [hunk 4]
```diff
@@ -7,7 +7,8 @@ <h5 class="modal-title" id="sync-dest-modal">Synchronization Destination Confirm
 </div>
 <div class="modal-body" id="modal-body">
     <div class="alert alert-warning" role="alert" th:text="#{screen.message.modal.preferences.warning}"></div>
-    <p th:utext="#{screen.message.common.syncDest.confirmationModalText}"></p>
+    <div>{{ confirmationMessage }}</div>
+    <b>{{ syncDestDescription }}</b>
```
Comment (JorWo) [inline src/main/resources/templates/modal/syncDestModal.html:11]: Could you rename the $scope variable confirmationMessage to something like syncDestConfirmationMessage to match the naming of $scope.syncDestDescription. Thanks!

File: src/main/resources/static/javascript/mainApp/grouping.controller.js (hunk 3 above)
Comment (yertsti) [reply src/main/resources/static/javascript/mainApp/grouping.controller.js:1589]: Angular complains about unknown provider if it isn't a function. I have no idea why isSynced works fine even though it isn't a function.

File: src/main/resources/templates/modal/syncDestModal.html (hunk 4 above)
Comment (JorWo) [reply src/main/resources/templates/modal/syncDestModal.html:11]: You may have forgotten to update `{{ confirmationMessage }}` to `{{ syncDestConfirmationMessage}}`.

File: src/main/resources/static/javascript/mainApp/grouping.controller.js (hunk 3 above)
Comment (JorWo) [reply src/main/resources/static/javascript/mainApp/grouping.controller.js:1589]: Maybe Angular doesn't like that $scope.selectedSyncDest.description is a $scope variable? Or maybe you need to wrap this resolve object in parenthesis? 
So possibly,
```
resolve: ({
    isSynced: isSyncDestOn,
    syncDestDescription: $scope.selectedSyncDest.description
})
```

Comment (yertsti) [reply src/main/resources/static/javascript/mainApp/grouping.controller.js:1589]: Still not sure why. I tried changing it to be a JS const and also wrapping it in parenthesis, but it still complaied about unknown provider

Comment (JorWo) [reply src/main/resources/static/javascript/mainApp/grouping.controller.js:1589]: That's ok, we can leave it as is.

**Review by JorWo — APPROVED** [review]
Great job!

**Pull request merged**